Things should be fixed now. I wasn't correctly following the `setup-uv` readme, which suggests doing e.g. `uv pip install --python=$PYTHON_VERSION pip` right after running the action with `python-version` set to ensure `pip` gets installed.

After fixing that it seems to fix the issues with using an incorrect Python install.

There was one other test failure I ran into on Python 3.11 that is unrelated to this PR but I saw in CI because uv installs the latest version of python 3.11 and setuptools. You can verify yourself that doing `import distutils.msvccompiler` fails in the `setuptools` monkeypatched version of `distutils`:

```bash
Python 3.11.11 (main, Jan 21 2025, 14:04:14) [Clang 16.0.0 (clang-1600.0.26.6)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from distutils import msvccompiler
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'msvccompiler' from 'distutils' (/Users/goldbaum/.pyenv/versions/3.11.11/lib/python3.11/site-packages/setuptools/_distutils/__init__.py)
>>> import setuptools
>>> setuptools.__version__
'75.8.0'
```


The fix is to use an older version of setuptools on Python < 3.12, which we were already doing in `test_requirements.txt`, but not in some of the CI setups. To fix that I created a new setuptools-specific requirements file. I could also just install the requirements in `test_requirements.txt` but I figured the CI jobs weren't use that for a reason (mypy maybe?).

--- a/.github/workflows/dependency-review.yml
+++ b/.github/workflows/dependency-review.yml
@@ -20,3 +20,5 @@ jobs:
           persist-credentials: false
       - name: 'Dependency Review'
         uses: actions/dependency-review-action@3b139cfc5fae8b618d3eae3675e383bb1769c019 # v4.5.0
+        with:
+          allow-ghsas: GHSA-cx63-2mw6-8hw5

